### ISSUE TITLE
(cli): --no-notices not reflected in synth

### ISSUE THREAD

**Issue by jaecktec** [aws/aws-cdk#19357]
### What is the problem?

When executing synth a call to `cli.cdk.dev-tools.aws.dev` is still being executed even though --no-notices was specified.
This causes builds in locked down environments to be super slow

### Reproduction Steps

```
CDK_HOME=$(mktemp -d) NODE_DEBUG=http npx cdk synth --no-notices
```

### What did you expect to happen?

log does not print 
```
HTTP 65935: createConnection cli.cdk.dev-tools.aws.dev:443::::::::::::::::::::: 
```
so no connection was established

### What actually happened?

log does print 
```
HTTP 65935: createConnection cli.cdk.dev-tools.aws.dev:443::::::::::::::::::::: 
```

### CDK CLI Version

2.15.0 (build 151055e)

### Node.js Version

2.15.0 (build 151055e)

### OS

macos

### Language

Typescript

### Language Version

2.15.0 (build 151055e)

### Other information

related to https://github.com/aws/aws-cdk/issues/19201

**Comment by peterwoodworth** [comment]
It seems to me that [the PR that closed the issue you linked](https://github.com/aws/aws-cdk/pull/19226) will resolve the issue you're experiencing once the next version comes out with that fix, since our latest version of v2 doesn't yet include that fix.

**Comment by jaecktec** [comment]
confirm it's fixed with 2.1.6
